pkg/operator/apiservicecontroller/apiservice_controller_test.go

```diff
@@ -153,24 +487,32 @@ func TestAvailableStatus(t *testing.T) {
 				kubeAggregatorClient.PrependReactor("*", "apiservices", tc.apiServiceReactor)
 			}
 
+			eventRecorder := events.NewInMemoryRecorder("")
 			fakeOperatorClient := operatorv1helpers.NewFakeOperatorClient(&operatorv1.OperatorSpec{ManagementState: operatorv1.Managed}, &operatorv1.OperatorStatus{}, nil)
 			operator := &APIServiceController{
-				apiServices: []*apiregistrationv1.APIService{
-					{
-						ObjectMeta: metav1.ObjectMeta{Name: "v1.apps.openshift.io"},
-						Spec:       apiregistrationv1.APIServiceSpec{Group: "apps.openshift.io", Version: "v1", Service: &apiregistrationv1.ServiceReference{}},
-					},
-					{
-						ObjectMeta: metav1.ObjectMeta{Name: "v1.build.openshift.io"},
-						Spec:       apiregistrationv1.APIServiceSpec{Group: "build.openshift.io", Version: "v1", Service: &apiregistrationv1.ServiceReference{}},
-					},
-				},
-				precondition:            func() (bool, error) { return true, nil },
+				precondition:            func([]*apiregistrationv1.APIService) (bool, error) { return true, nil },
 				kubeClient:              kubeClient,
-				eventRecorder:           events.NewInMemoryRecorder(""),
+				eventRecorder:           eventRecorder,
 				operatorClient:          fakeOperatorClient,
 				apiregistrationv1Client: kubeAggregatorClient.ApiregistrationV1(),
 				versionRecorder:         status.NewVersionGetter(),
+				getAPIServicesToManageFunc: GetAPIServicesToManageWrapper(
```

Comment:
Yeah, a separate struct with a `NewAPIServicesToManage` function is likely to read easier.